src/main/java/Main.java

```diff
@@ -41,6 +44,33 @@ public static void producerRecordTest(){
         System.out.println("TEST 6");
         System.out.println(test6 + "\n");
 
+        System.out.println("Properties class main testing ");
+
+        Map<String, String> initialProperties = new HashMap<>();
+        initialProperties.put("retries", "3");
+        initialProperties.put("timeout", "5000");
+
+        Properties producerProperties = new Properties(initialProperties);
+
+        producerProperties.setProperty("yo mama", "max");
+
+        System.out.println("Retries: " + producerProperties.getProperty("retries"));
+        System.out.println("Timeout: " + producerProperties.getProperty("timeout"));
+
+        // Handling the case where 'acknowledgment' does not exist
+        String acknowledgment = producerProperties.getProperty("acknowledgment");
+        if (acknowledgment != null) {
+            System.out.println("Acknowledgment: " + acknowledgment);
+        } else {
+            System.out.println("Acknowledgment: Not set");
```

Comment:
Technically it's possible that the a key can have a value of `null`

Reply:
Do you think I should approach this by making sure every value will have a default value or an expection.

Reply:
Up to you, I'm down either way. The way you currently have it written looks good to me, i was just pointing out that `null` technical != "Not set"